src/explorer.ts

```diff
@@ -230,14 +230,76 @@ class KubernetesDeploymentResource extends KubernetesResource {
         super(kind, id, metadata);
     }
 
-    getTreeItem(): vscode.TreeItem | Thenable<vscode.TreeItem> {
-        const treeItem = new vscode.TreeItem(this.id, vscode.TreeItemCollapsibleState.Collapsed);
-        treeItem.contextValue = `vsKubernetes.resource.${this.kind.abbreviation}`;
+    async getTreeItem(): Promise<vscode.TreeItem> {
+        const treeItem = await super.getTreeItem();
+        treeItem.collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
         return treeItem;
     }
 
     async getChildren(kubectl: Kubectl, host: Host): Promise<KubernetesObject[]> {
         const pods = await kubectlUtils.getPods(kubectl, this.metadata.selector);
         return pods.map((p) => new KubernetesResource(kuberesources.allKinds.pod, p.name, p));
     }
-}
\ No newline at end of file
+}
+
+class KubernetesDataHolderFolder extends KubernetesResourceFolder {
+    readonly resource: string;
+    constructor(resource: string, kind: kuberesources.ResourceKind) {
+        super(kind);
+        this.resource = resource;
```

Comment:
I believe that in TypeScript you can write `constructor(private readonly resource: string, ...)` and it will create and assign the `readonly resource` field for you.  https://www.typescriptlang.org/docs/handbook/classes.html

Reply:
done.